Phoenix/Core/Source/Voxels/Blocks.cpp

```diff
@@ -38,28 +36,26 @@ RegisteredBlock::RegisteredBlock(const std::string& unique, int id,
 
 RegisteredBlock::~RegisteredBlock() = default;
 
-BlockRegistry::BlockRegistry() : i(0) {};
-
-BlockRegistry::~BlockRegistry() {};
+BlockRegistry::BlockRegistry() : m_BlockIdCounter(0) {};
 
 int BlockRegistry::registerBlock(const std::string& uniqueName,
                                  const std::string& displayName)
 {
-	i++;
-	Blocks[i] = RegisteredBlock(uniqueName, i, displayName);
-	return i;
+	m_BlockIdCounter++;
+	m_Blocks.push_back(RegisteredBlock(uniqueName, m_BlockIdCounter, displayName));
+	return m_BlockIdCounter;
 };
 
 const std::string& BlockRegistry::getDisplayName(int blockId)
 {
-	return Blocks[blockId].displayName;
+	return m_Blocks[blockId].displayName;
 };
 
 int BlockRegistry::getBlockId(const std::string& uniqueName)
 {
-	for (i = 0; i < 100; i++)
+	for (int i = 0; i < 100; i++)
```

Comment:
a small nitpick - I'd use `std::size_t` here over int.

Also why are you iterating to 100? Should this not iterate over all the elements in the vector as this will not work if there are less than 100 items in the vector and the uniqueName does not exist in the vector.

Reply:
Ahh, back when the vector was an array this made sense. Updated to match the vector.